examples/material_gallery/lib/gallery/demo.dart

```diff
@@ -0,0 +1,252 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_markdown/flutter_markdown.dart';
+
+import '../gallery/syntax_highlighter.dart';
+
+class SingleComponentDemoData {
+  SingleComponentDemoData({
+    this.builder,
+    this.exampleCode,
+    this.description,
+    this.onPressedDemo
+  });
+
+  final WidgetBuilder builder;
```

Comment:
I felt like the build methods where each example is a separate function made the code easier to read for beginner flutter users.